bigquery/gcloud/aio/bigquery/bigquery.py

```diff
@@ -24,13 +34,13 @@ class Table:
     def __init__(self, dataset_name: str, table_name: str,
                  project: Optional[str] = None,
                  service_file: Optional[Union[str, io.IOBase]] = None,
-                 session: Optional[aiohttp.ClientSession] = None,
+                 session: Optional[Session] = None,
```

Comment:
Hm, why is `session` optional in this class, but not in the `IamClient` class?

Reply:
From what I can tell, the it is probably something to do with the way the libraries are expected to be used. `auth` package seems to be a linked dependency for every other `gcloud` package and probably the original design had it as a required dependency to force the calling package to define a `session` so that the same `session` can also be used by the caller.

Which now makes me think if we should actually allow it as an optional parameter now in the updated code because this may create 2 sessions.

We can alternately update the code to pass the new `session` created in this line:
```self.session = RestSession(session) if session else RestSession()``` to other classes like `Token` like ```Token(..., session=self.session.session```

Thoughts?

Reply:
Interesting dependency... And I thought based on your comment above that the `IamClient` would have an optional `session` input, but it's now required (but set to be `Optional`) as a positional argument.

Reply:
Crap, my bad, I am making it optional now.